controllers/hosting/model_reconciler.go

```diff
@@ -392,12 +392,17 @@ func (r *modelReconciler) getContainerDefinitions(model *commonv1.Model, contain
 // This also returns an error if the primary container definition cannot be found.
 func (r *modelReconciler) getPrimaryContainerDefinition(model *commonv1.Model, containers map[string]*commonv1.ContainerDefinition) (*commonv1.ContainerDefinition, error) {
 
+	// For multi-container, don't expect a primary container
+	if model.PrimaryContainer == nil && len(model.Containers) > 1 {
+		return nil, nil
+	}
+
 	// Get primary container name.
 	// If none is specified and the list of required containers has only one element, use that container as the primary.
 	var primaryContainerName string
 	if model.PrimaryContainer != nil {
 		primaryContainerName = *model.PrimaryContainer
-	} else if len(model.Containers) == 1 {
+	} else if len(model.Containers) > 0 {
 		primaryContainerName = *model.Containers[0].ContainerHostname
 	} else {
```

Comment:
why do we need this "else" ?

Reply:
Although we have a client-side validation for if `len(containers) > 0`, that condition just double checks that this is true. It was one of the conditions of the unit tests.

Reply:
no, i'm talking abt the last else 

```
else {
 return nil, fmt.Errorf("Unable to determine primary ... Either specify explicitly or provide only one container")
}
```

Reply:
Yes that's is also what I'm referring to. In the case that for whatever reason we have no containers, it is a fallback - otherwise we will be doing a null pointer dereference.

Reply:
in that case can u edit the error message to something like "no containers specified"